src/util/system.cpp

```diff
@@ -17,16 +17,6 @@
 #endif
 
 #ifndef WIN32
-// for posix_fallocate
-#ifdef __linux__
-
-#ifdef _POSIX_C_SOURCE
-#undef _POSIX_C_SOURCE
-#endif
-
-#define _POSIX_C_SOURCE 200112L
-
-#endif // __linux__
```

Comment:
I think we shouldn't remove this. We are setting `_POSIX_C_SOURCE` in order to ask the system to provide the functions. [feature_test_macros(7)](https://linux.die.net/man/7/feature_test_macros) explains that indeed we are supposed to set it ourselves (not just observe its default value). And, of course, set it before including the system headers.

Further, we should consider OSes that will not provide the functions we need (`posix_fadvise()` in this case) even if we define `_POSIX_C_SOURCE` to `200112` (e.g. MacOS). So, a [configure-time check](https://github.com/bitcoin/bitcoin/pull/18437/files#diff-67e997bcfdac55191033d57a16d1408aR854-R867) is warranted, similarly to what https://github.com/bitcoin/bitcoin/pull/18437 does.